src/viking.js

```diff
@@ -1,14 +1,98 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength) {
+    this.health = health;
+    this.strength = strength;
+  }
+
+  attack() {
+    return this.strength;
+  };
+
+  receiveDamage(theDamage) {
+    this.health = this.health-theDamage;
+  };
+
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier{
+  constructor(name, health, strength) {
+    super(health, strength);
+    this.name = name;
+  }
+  receiveDamage(theDamage){
+    this.health = this.health-theDamage;
+    if(this.health>0) {
+      return `${this.name} has received ${theDamage} points of damage`;
+    } else {
+      return `${this.name} has died in act of combat`;
+    }
+  };
+  battleCry(){
+      return (`Odin Owns You All!`);
+    };
+    
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier{
+  constructor(health, strength){
+    super(health,strength);
+  }
+  receiveDamage(theDamage) {
+    this.health = this.health-theDamage;
+    if(this.health>0) {
+      return `A Saxon has received ${theDamage} points of damage`;
+    } else {
+      return `A Saxon has died in combat`;
+    }
+  };
+}
 
 // War
-class War {}
+class War {
+    vikingArmy = [];
+    saxonArmy = [];
+    addViking(viking) {
+      this.vikingArmy.push(viking);
+    };
+    addSaxon(saxon){
+      this.saxonArmy.push(saxon);
+    };
+    vikingAttack() {
+      let randomSaxonSoldier = Math.floor(Math.random() * this.saxonArmy.length);
+      let randomSaxon = this.saxonArmy[randomSaxonSoldier];
+      let randomViking = this.vikingArmy[Math.floor(Math.random() * this.vikingArmy.length)];
+      let resultVikingAttacks = randomSaxon.receiveDamage(randomViking.strength);
+
+      if (randomSaxon.health <= 0) {
+        this.saxonArmy.splice(randomSaxonSoldier,1);
+      }
+      return resultVikingAttacks;
+    };
+    saxonAttack(){
+      let randomVikingSoldier = Math.floor(Math.random() * this.vikingArmy.length);
+      let randomViking = this.vikingArmy[randomVikingSoldier];
+      let randomSaxon = this.SaxonArmy[Math.floor(Math.random() * this.saxonArmy.length)];
```

Comment:
Ele deixou de passar em 4 itens do teste, porque o S do this.SaxonArmy está maiúsculo. Arrumando isso ele passa em tudo! 
Muito bom, parabéns!